cpp/inc/bond/ext/grpc/detail/service_call_data.h

```diff
@@ -81,30 +81,32 @@ struct service_unary_call_data : io_manager_tag
     {
         if (ok)
         {
-            // capture the data associated with this one incomming request
-            // so that we can pass it to the user callback. The unary_call
-            // that we create to pass to the user callback takes ownership
-            // of this data.
-            //
-            // This scope block is here so that the shared_ptr that we
-            // create to capture in the lambda is destroyed after we enqueue
-            // in the thread pool but before we schedule the next receive.
-            // When we can use C++14, we can simplify this by using lambda's
-            // capture-by-move.
+            // Capture the data associated with this one incomming request
+            // so that we can pass it to the user callback. When we create
+            // the unary_call to pass to the user callback, the unary_call
+            // and unary_call_impl start collaborating to manage the
+            // lifetime of the unary_call_impl.
             {
-                std::shared_ptr<unary_call_impl<TRequest, TResponse>> receivedCall = _receivedCall;
+                // We have to manually release the pointer from
+                // _receivedCall to capture it in the lambda. When we can
+                // use C++14, we can simplify this by using lambda's
+                // capture-by-move.
+                unary_call_impl<TRequest, TResponse>* receivedCall = _receivedCall.release();
 
                 _threadPool->schedule([this, receivedCall]() mutable
                 {
                     _cb(unary_call<TRequest, TResponse> { std::move(receivedCall) });
-
-                    // should have transfered ownership
-                    BOOST_ASSERT(!receivedCall);
                 });
+
+                // If _threadPool->schedule throws an exception, we don't
+                // delete the receivedCall that was just released; however
+                // that exception is going to leak into io_manager, which
+                // can't handle unhandled exceptions, and will terminate the
+                // program.
             }
 
             // create new state for the next request that will be received
-            _receivedCall = std::make_shared<unary_call_impl<TRequest, TResponse>>();
+            _receivedCall.reset(new unary_call_impl<TRequest, TResponse>);
```

Comment:
`make_unique` here as well?

Reply:
`std::make_unique` was only added in C++14.

Reply:
Oh, right.